src/packages.builds

```diff
@@ -15,6 +15,40 @@
     </Project>
   </ItemGroup>
 
+  <!-- Need the PackageIndexFile file property from baseline.props -->
+  <Import Project="../pkg/baseline/baseline.props" />
+
+  <UsingTask TaskName="UpdatePackageIndex" AssemblyFile="$(PackagingTaskDir)Microsoft.DotNet.Build.Tasks.Packaging.dll"/>
+
+  <!--
+    Updates the package index to marke all packages we are building that can go stable as stable.
+    this will allow for a kicked off build to control package stability at queue time. This does edit
+    the package index in-place but that shouldn't cause any problems for official builds are the only
+    onces that might do this. After we ship a stable set of packages this target should be ran and the
+    changes to the package index should be commited to the repo.
+  -->
+  <Target Name="GeneratePackageIndexWithStableVersions"
+          BeforeTargets="Build"
+          Condition="'$(IncludePreReleaseLabelInPackageVersion)' != 'true'">
+    <ItemGroup>
+      <PkgProjects Include="@(Project)" Condition="'%(Project.Extension)' == '.pkgproj'" />
+    </ItemGroup>
+
+    <MSBuild Targets="GetPackageIdentityIfStable"
+             BuildInParallel="$(BuildInParallel)"
+             Projects="@(PkgProjects)">
```

Comment:
Do we ensure that we have a flat list (including runtime split packages)?  I think we do, just double checking.

Reply:
I actually had to change this to just pull in all pkgprojs and by-pass the builds files, so there is going to be some slight duplication in declaring the file list but that shouldn't be a big deal.

Reply:
I see, I missed that itemgroup above.  I don't think this will work for pkgproj's that require additional properties to calculate their ID (like the RID-split framework ones).  That may not be an issue here where we aren't shipping any RID-split packages as stable, but it will be in core-setup.

Reply:
We don't use a package index in core-setup so we don't have this problem.